core/domain/classifier_services_test.py

```diff
@@ -92,6 +92,103 @@ def test_string_classifier_classification(self):
             self.assertTrue(
                 self._is_string_classifier_called('dunno, just guessed'))
 
+    def test_check_re_training_conditions(self):
+        """Test the check_re_training_conditions method."""
+        exploration = exp_services.get_exploration_by_id(self.exp_id)
+        state = exploration.states['Home']
+        training_data = state.get_training_data()
+        interaction_id = state.interaction.id
+        algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING[interaction_id][
+            'algorithm_id']
+        job_id = classifier_services.create_classifier_training_job(
+            algorithm_id, interaction_id, exploration.id, exploration.version,
+            'Home', training_data, feconf.TRAINING_JOB_STATUS_NEW)
+
+        # Test change in number of answer groups.
+        state.interaction.answer_groups.insert(
+            3, state.interaction.answer_groups[1])
+        self.assertTrue(classifier_services.check_re_training_conditions(
+            state, job_id))
+
+        # Test change in number of samples more than MIN_SAMPLE_TO_RETRAIN.
+        state = exploration.states['Home']
+        for (answer_group_index, answer_group) in enumerate(
+                state.interaction.answer_groups):
+            classifier_rule_spec_index = (
+                answer_group.get_classifier_rule_index())
+            if classifier_rule_spec_index is not None:
+                extra_training_data = ['someval'] * 20
+                state.interaction.answer_groups[answer_group_index].rule_specs[
+                    0].inputs['training_data'].extend(extra_training_data)
+        self.assertTrue(classifier_services.check_re_training_conditions(
+            state, job_id))
+
+    def test_add_to_queue(self):
+        """Test the add_to_training_queue method."""
+        exploration = exp_services.get_exploration_by_id(self.exp_id)
+        state = exploration.states['Home']
+        classifier_services.add_to_training_queue(exploration)
+
+        # Since the exploration has only one trainable state, there should be
+        # only one job model in data store.
+        all_jobs = classifier_models.ClassifierTrainingJobModel.get_all()
+        job_num = 0
+        for job in all_jobs:
```

Comment:
perhaps - first assert that `all_jobs` has only one job by `self.assertEqual(len(all_jobs), 1)` and then when you have asserted that all_jobs has just one element, get its id.